py/sentry_relay/processing.py

```diff
@@ -262,3 +263,25 @@ def validate_project_config(config, strict: bool):
     error = decode_str(raw_error, free=True)
     if error:
         raise ValueError(error)
+
+
+def run_dynamic_sampling(sampling_config, root_sampling_config, dsc, event):
+    """
+    Runs dynamic sampling on an event and returns the merged rules together with the sample rate.
+    """
+    assert isinstance(sampling_config, string_types)
+    assert isinstance(root_sampling_config, string_types)
+    assert isinstance(dsc, string_types)
+    assert isinstance(event, string_types)
+
+    return json.loads(
+        decode_str(
+            rustcall(
+                lib.run_dynamic_sampling,
+                encode_str(sampling_config),
+                encode_str(root_sampling_config),
+                encode_str(dsc),
+                encode_str(event),
+            )
+        )
+    )
```

Comment:
Please add `free=True` to `decode_str`, since the owned string returned by FFI needs to be released. You can make this more readable like so:

```suggestion
    result_json = rustcall(
        lib.run_dynamic_sampling,
        encode_str(sampling_config),
        encode_str(root_sampling_config),
        encode_str(dsc),
        encode_str(event),
    )

    return json.loads(decode_str(result_json, free=True))
```